modules/auxiliary/server/capture/ldap.rb

```diff
@@ -0,0 +1,249 @@
+        ],
+        'PassiveActions' => [ 'Capture' ],
+        'DefaultActions' => 'Capture',
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptAddress.new('SRVHOST', [ true, 'The localhost to listen on.', '0.0.0.0' ]),
+        OptPort.new('SRVPORT', [ true, 'The local port to listen on.', '389' ])
+      ]
+    )
+
+    deregister_options('LDIF_FILE')
+
+    register_advanced_options(
+      [
+        OptPath.new('LDIF_FILE', [ false, 'Directory LDIF file path'])
+      ]
+    )
+  end
+
+  def run
+    exploit
+  end
+
+  def on_dispatch_request(client, data)
+    return if data.strip.empty? || data.strip.nil?
+
+    state = {}
+
+    state[client] = {
+      name: "#{client.peerhost}:#{client.peerport}",
+      ip: client.peerhost,
+      port: client.peerport
+    }
+
+    data.extend(Net::BER::Extensions::String)
+    begin
+      pdu = Net::LDAP::PDU.new(data.read_ber!(Net::LDAP::AsnSyntax))
+      vprint_status("LDAP request data remaining: #{data}") unless data.empty?
+
+      res = case pdu.app_tag
+            when Net::LDAP::PDU::BindRequest
+              domains = []
+              names = []
+              result_code = nil
+              user_login = pdu.bind_parameters
+
+              if user_login.name.empty? || user_login.authentication.empty?
+                state[client][:user] = user_login.name
+                state[client][:pass] = user_login.authentication
+                state[client][:domain] = nil
+                result_code = Net::LDAP::ResultCodeSuccess
+              elsif !user_login.name.empty?
+                if user_login.name =~ /@/
+                  pub_info = user_login.name.split('@')
+                  if pub_info.length <= 2
+                    state[client][:user] = pub_info[0]
+                    state[client][:domain] = pub_info[1]
+                  else
+                    result_code = Net::LDAP::ResultCodeInvalidCredentials
+                    print_error("LDAP Login Attempt => From:#{state[client][:name]} DN:#{user_login.name}")
+                  end
+                elsif user_login.name =~ /,/
+                  begin
+                    dn = Net::LDAP::DN.new(user_login.name)
+                    dn.each_pair do |key, value|
+                      if key == 'cn'
+                        names << value
+                      elsif key == 'dc'
+                        domains << value
+                      end
+                    end
+                    state[client][:user] = names.first
+                    state[client][:domain] = domains.empty? ? nil : domains.join('.')
+                  rescue StandardError => e
+                    print_error("LDAP Login Attempt => From:#{state[client][:name]} DN:#{user_login.name}")
+                    raise e
+                  end
+                else
+                  state[client][:user] = ''
```

Comment:
This is branch of the conditional us when a plain the username was provided without `@` or `.` and should be reported as such not set to an empty string.
```suggestion
                  state[client][:user] = user_login.name
```

Another enhancement that may be appropriate here would be to support when a `NETBIOS` formatted user is used. Example:
```
EXAMPLEDOMAIN\username
```

Reply:
Please check if the adjustment is acceptable